### PR TITLE
openthread: optimize FTD and MTD configurations

### PR DESCRIPTION
**Pull request by maciejbaczmanski**
Removed support for:
- DHCPv6 client
- Joiner
- SNTP client
- Link Metrics initiator

in FTD and MTD variants to optimize space for application.

### CODE REVIEW

**Review by LuDuda — APPROVED** [review]

File: openthread/Kconfig [hunk 1]
```diff
@@ -81,16 +75,12 @@ config OPENTHREAD_NORDIC_LIBRARY_FTD
...
 	select OPENTHREAD_EXTERNAL_HEAP
 	select OPENTHREAD_IP6_FRAGM
-	select OPENTHREAD_JOINER
 	select OPENTHREAD_NETDIAG_CLIENT
 	select OPENTHREAD_SLAAC
-	select OPENTHREAD_SNTP_CLIENT
 	select OPENTHREAD_SRP_CLIENT
-	select OPENTHREAD_DUA if !OPENTHREAD_THREAD_VERSION_1_1
 	select OPENTHREAD_MLR if !OPENTHREAD_THREAD_VERSION_1_1
 	select OPENTHREAD_LINK_METRICS_INITIATOR if !OPENTHREAD_THREAD_VERSION_1_1
```
Comment (edmont) [inline openthread/Kconfig:85]: I believe we can remove `OPENTHREAD_LINK_METRICS_INITIATOR` from FTD as well, the requirement is to support the Subject role only.

File: openthread/Kconfig [hunk 2]
```diff
@@ -81,16 +75,12 @@ config OPENTHREAD_NORDIC_LIBRARY_FTD
...
-	select OPENTHREAD_DHCP6_CLIENT
 	select OPENTHREAD_ECDSA
 	select OPENTHREAD_EXTERNAL_HEAP
 	select OPENTHREAD_IP6_FRAGM
-	select OPENTHREAD_JOINER
 	select OPENTHREAD_NETDIAG_CLIENT
 	select OPENTHREAD_SLAAC
-	select OPENTHREAD_SNTP_CLIENT
 	select OPENTHREAD_SRP_CLIENT
-	select OPENTHREAD_DUA if !OPENTHREAD_THREAD_VERSION_1_1
```
Comment (edmont) [inline openthread/Kconfig:93]: Please make sure Thread 1.2 Certification 5.9.x tests are optional.

**Review by edmont — COMMENTED** [review]
Not sure about DUA being required or not.

File: openthread/Kconfig (hunk 1 above)
Comment (maciejbaczmanski) [reply openthread/Kconfig:85]: @LuDuda , if I remember correctly, you were saying that support for `OPENTHREAD_LINK_METRICS_INITIATOR` should stay in FTD.  Do you think we can remove it then or are there some other reasons to leave it?

**Review by canisLupus1313 — APPROVED** [review]

File: openthread/Kconfig (hunk 2 above)
Comment (maciejbaczmanski) [reply openthread/Kconfig:93]: We'll discuss that on today's meeting with GRL

Comment (maciejbaczmanski) [reply openthread/Kconfig:93]: Discussion on DUA has been brought up to Thread Group, but will be continued separately. Reverted removing `OPENTHREAD_DUA` from this PR.

File: openthread/Kconfig (hunk 1 above)
Comment (LuDuda) [reply openthread/Kconfig]: I think we can remove it for FTD as well if its not required. I was thinking more about Initiator role in general.

**Review by edmont — APPROVED** [review]

File: openthread/Kconfig (hunk 1 above)
Comment (maciejbaczmanski) [reply openthread/Kconfig]: removed

**Pull request merged**